From 662f121947ca7c5196ae3f97a0c43b788f8da23e Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 21 Sep 2017 11:44:38 +0800
Subject: [PATCH 2581/5242] MLK-16536-7 video: fbdev: dcss: abstract cfifo wq
 flush operation

commit  7ae5e561aa236fee7900d3477786efb3735ce5fb from
https://source.codeaurora.org/external/imx/linux-imx.git

Abstract the cfifo workqueue flush operation to a separate
interface 'finish_cfifo()'.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 842f249..30ad004 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2342,6 +2342,13 @@ static void flush_cfifo(struct ctxld_fifo *cfifo,
 	WARN(!ret, "work has already been queued\n");
 }
 
+static int finish_cfifo(struct ctxld_fifo *cfifo)
+{
+	flush_workqueue(cfifo->ctxld_wq);
+
+	return 0;
+}
+
 static int commit_cfifo(uint32_t channel,
 			struct dcss_info *info,
 			struct ctxld_commit *cc)
@@ -2380,7 +2387,7 @@ static int commit_cfifo(uint32_t channel,
 		atomic_set(&info->flush, 1);
 		spin_unlock(&cfifo->cqueue.lock);
 		/* Wait fifo flush empty to avoid fifo wrap */
-		flush_workqueue(cfifo->ctxld_wq);
+		finish_cfifo(cfifo);
 		spin_lock(&cfifo->cqueue.lock);
 		atomic_set(&info->flush, 0);
 		kfifo_reset(&cfifo->fifo);
@@ -2788,7 +2795,7 @@ static int dcss_pan_display(struct fb_var_screeninfo *var,
 	/* TODO: blocking mode */
 	if (likely(!var->reserved[2]))
 		/* make pan display synchronously */
-		flush_workqueue(info->cfifo.ctxld_wq);
+		finish_cfifo(&info->cfifo);
 
 	goto out;
 
-- 
1.7.9.5

